### PR TITLE
USAGOV-1594-tome-basepath: On export, Check basepath to exclude configured paths

### PR DESCRIPTION
**Pull request by omerida**
When exporting to a subdirectory, account for the base path when we test if a directory should be excluded from export.

## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-1594

## Description
When testing if a path should be exported or excluded, adds a check to see if the path starts with the basepath when tome is exporting into a subdirectory of the document root.

## Type of Changes

- [ ] New Feature
- [x] Bugfix
- [ ] Frontend (Twig, Sass, JS)
  - Add screenshot showing what it should look like
- [ ] Drupal Config (requires "drush cim")
- [ ] New Modules (requires rebuild)
- [ ] Infrastructure
  - [ ] CMS
  - [ ] WAF
  - [ ] Egress
  - [ ] Tools
- [ ] Other

## Testing Instructions

Export the site with a URI like:

```
drush tome:static --process-count=80 --uri="https://static.example.com/static/export/"
```

Your export folder should not have files from directories in `$settings['usagov_tome_static_path_exclude_directories']`. You should not see a `html/static/export/node` directory, `html/static/export/node` directory, `html/static/export/es/node` directory, `html/static/export/jsonapi` directory, etc.

### Requires New Config
- [ ] Yes
- [x] No

### Requires New Content
- [ ] Yes
- [x] No

### Validation Steps
- Test instruction 1
- Test instruction 2
- Test instruction 3

## Security Review

- [ ] Adds/updates software (including a library or Drupal module)
- [ ] Communication with external service
- [ ] Changes permissions or workflow
- [ ] Requires SSPP updates

## Reviewer Reminders
- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions
Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.

### CODE REVIEW

File: web/modules/custom/usagov_ssg_postprocessing/src/EventSubscriber/TomeEventSubscriber.php [hunk 1]
```diff
@@ -91,14 +91,16 @@ public function excludeDirectories(CollectPathsEvent $event) {
           $paths[$path] = $metadata;
         }
       }
-
+      // Get the base path without trailing slash if we're exporting to a sub-directory
+      $base_path = rtrim(trim(base_path()), '/');
       foreach ($excluded_directories as $excluded_directory_path) {
         $excluded_directory = $excluded_directory_path . '/';
         if (($path == $excluded_directory_path) ||
```
Comment (akf) [inline web/modules/custom/usagov_ssg_postprocessing/src/EventSubscriber/TomeEventSubscriber.php:98]: I wound up with two unexpected paths in my export: 
```
.../static/export/jsonapi/index.html
.../static/export/paragraphs_entity_embed/autocomplete/index.html
```
I suspect, but did not verify, that this clause should be expanded like the one on line 103 to catch those.

**Review by akf — CHANGES_REQUESTED** [review]
I caught a couple of missed files in my exported subdirectory. I think I know why (see comment). 

I also wound up with a bunch of directories that landed "outside" my subdirectory. I don't usually export to a subdirectory, so maybe this is a known issue that just wasn't known to me? (If so, obviously you don't need to fix it!) I'm going to revert to the dev branch and start another export and check it in the morning. 

I'm exporting into html, which is our stock setup. Specifying `--uri="https://static.example.com/static/export/"` as in your example, I wound up with 425 directories under `html/static/export` and 72 under `html`. As an example, `where-report-scam` is in both, with different contents: 

```
amykfarrell@FCOH2J-T6L7Q05P usagov-2021 % ls html/where-report-scam
identity-theft				romance
imposter				what-type-scam-do-you-need-report
index.html
amykfarrell@FCOH2J-T6L7Q05P usagov-2021 % ls html/static/export/where-report-scam
imposter				what-type-scam-do-you-need-report
```

**Review by akf — COMMENTED** [review]
Update: my tome run on the dev branch also produced a bunch of pages under html/ instead of html/static/export, so that appears to have nothing to do with this change.

File: web/modules/custom/usagov_ssg_postprocessing/src/EventSubscriber/TomeEventSubscriber.php (hunk 1 above)
Comment (omerida) [reply web/modules/custom/usagov_ssg_postprocessing/src/EventSubscriber/TomeEventSubscriber.php:98]: Yes, I missed testing the directory paths. Latest commit fixes this.

**Comment by omerida** [comment]
> Update: my tome run on the dev branch also produced a bunch of pages under html/ instead of html/static/export, so that appears to have nothing to do with this change.

I dug a bit into this and it seems to be a bug in tome itself, in `tome_static\src\EventSubscriber\RedirectPathSubscriber.php`. Seems like it should prepend the base path when the export is within a directory.

File: web/modules/custom/usagov_ssg_postprocessing/src/EventSubscriber/TomeEventSubscriber.php (hunk 1 above)
Comment (omerida) [reply web/modules/custom/usagov_ssg_postprocessing/src/EventSubscriber/TomeEventSubscriber.php:98]: I did a local export and didn't see those directories in the export.

**Review by akf — APPROVED** [review]

**Pull request merged**